Core/Core/Transports/Memory.cs

```diff
@@ -107,18 +104,18 @@ public void SaveObject(string id, ITransport sourceTransport)
 
   public Task WriteComplete()
   {
-    return Utilities.WaitUntil(() => true);
+    return Task.CompletedTask;
```

Comment:
Is this a bit naughty? Or is it just always complete because the act of saving alwaygs completes is the contract for this such that you write and wait for complete and then write the next blob and so on?